File: atcoder/abc132A_fifty_fifty_.hs
-- unihernandez22
-- https://atcoder.jp/contests/abc132/tasks/abc132_a
-- greedy

import Data.Set (fromList, size)

main = do
  s <- getLine
  putStrLn $
    if (size $ fromList s) == 2 then
      "Yes"
    else
      "No"

File: atcoder/abc132A_fifty_fifty.ml
(* vicfred
 * https://atcoder.jp/contests/abc132/tasks/abc132_a
 * implementation
 * *)
Scanf.scanf "%c%c%c%c\n" @@ fun a b c d ->
    print_endline @@
    if a = b && c = d && a != c ||
    a = c && b = d && a != b ||
    a = d && b = c && a != b then "Yes"
    else "No"

